### DISCUSSION TITLE
how to remove watermark

### DISCUSSION THREAD

**Discussion by DarynKash** [aclap-dev/video-downloadhelper#1578] [Help and Q&A]
### Your question

I just purchased the license for my downloader, can i use my license now to remove watermark now??

### Addon & CoApp version and details

not sure what to write here

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

You have some videos you downloaded with VDH before you got a license & those videos have watermarks.  To write the watermark VDH reads each frame of the video, alters it by inserting the watermark, then writes the new frame into the output file.  This should explain why writing the watermark takes a rather long time.  The longer the video, the more frames, so the more frames VDH alters.

But this is inherently a destructive process.  Whatever video content was in the location of the watermark is obliterated when the watermark is written.  You can't get it back.  There is no way to erase a watermark once it is in a video.  If a program existed that erased the watermark -- and to my knowledge no such program does exist -- you would just have a black hole (or some other color or pattern) where the watermark had been.  That's hardly better that having the watermark.

The best you can do is download the same content again now that you have a license.  And I know there are cases in which the content might no longer be available.  But if it is still available, you just have to download it again.  This new copy would not have the watermark.  This is the single effect of getting a license.  (Well, there are other minor effects in certain cases, but this is the main effect.)  Without a license, VDH writes the watermark.  With a license, VDH skips writing the watermark.  The license does not unlock certain functions in VDH.  The license does not make VDH work on more web sites.  VDH functions exactly the same with or without a license, except for the writing (or not) of the watermark.
